crmd/notify.c

```diff
@@ -20,120 +20,501 @@
 #include <crm/crm.h>
 #include <crm/msg_xml.h>
 #include "notify.h"
+#include "crmd_messages.h"
 
 char *notify_script = NULL;
 char *notify_target = NULL;
+GListPtr notify_list = NULL;
+
+typedef struct {
+    char *id;
+    char *path;
+    int timeout;
+    char *tstamp_format;
+    char *info;
```

Comment:
For the schema, see e.g. xml/resources-1.3.rng (element-resource-extra) and xml/nvset-1.3.rng. For the code, see lib/pengine/rules.c:unpack_instance_attributes() and uses of it.